src/Datadog.Trace.ClrProfiler.Native/calltarget_tokens.cpp

```diff
@@ -0,0 +1,2031 @@
+    auto* signature = new COR_SIGNATURE[signatureLength];
+    unsigned offset = 0;
+
+    signature[offset++] = IMAGE_CEE_CS_CALLCONV_DEFAULT;
+    signature[offset++] = 0x00;
+
+    signature[offset++] = ELEMENT_TYPE_VALUETYPE;
+    memcpy(&signature[offset], &callTargetReturnVoidTypeBuffer,
+           callTargetReturnVoidTypeSize);
+    offset += callTargetReturnVoidTypeSize;
+
+    hr = module_metadata->metadata_emit->DefineMemberRef(
+        callTargetReturnVoidTypeRef,
+        managed_profiler_calltarget_returntype_getdefault_name.data(),
+        signature, signatureLength, &callTargetReturnVoidTypeGetDefault);
+    if (FAILED(hr)) {
+      Warn("Wrapper callTargetReturnVoidTypeGetDefault could not be defined.");
+      return mdMemberRefNil;
+    }
+  }
+
+  return callTargetReturnVoidTypeGetDefault;
+}
+
+mdMemberRef CallTargetTokens::GetCallTargetReturnValueDefaultMemberRef(
+    mdTypeSpec callTargetReturnTypeSpec) {
+  auto hr = EnsureBaseCalltargetTokens();
+  if (FAILED(hr)) {
+    return mdMemberRefNil;
+  }
+  if (callTargetReturnTypeRef == mdTypeRefNil) {
+    Warn("Wrapper callTargetReturnTypeGetDefault could not be defined because callTargetReturnTypeRef is null.");
+    return mdMemberRefNil;
+  }
+
+  mdMemberRef callTargetReturnTypeGetDefault = mdMemberRefNil;
+
+  // *** Ensure CallTargetReturn<T>.GetDefault() member ref
+  ModuleMetadata* module_metadata = GetMetadata();
+
+  unsigned callTargetReturnTypeRefBuffer;
+  auto callTargetReturnTypeRefSize = CorSigCompressToken(
+      callTargetReturnTypeRef, &callTargetReturnTypeRefBuffer);
+
+  auto signatureLength = 7 + callTargetReturnTypeRefSize;
+  auto* signature = new COR_SIGNATURE[signatureLength];
+  unsigned offset = 0;
+
+  signature[offset++] = IMAGE_CEE_CS_CALLCONV_DEFAULT;
+  signature[offset++] = 0x00;
+  signature[offset++] = ELEMENT_TYPE_GENERICINST;
+  signature[offset++] = ELEMENT_TYPE_VALUETYPE;
+  memcpy(&signature[offset], &callTargetReturnTypeRefBuffer,
+         callTargetReturnTypeRefSize);
+  offset += callTargetReturnTypeRefSize;
+  signature[offset++] = 0x01;
+  signature[offset++] = ELEMENT_TYPE_VAR;
+  signature[offset++] = 0x00;
+
+  hr = module_metadata->metadata_emit->DefineMemberRef(
+      callTargetReturnTypeSpec,
+      managed_profiler_calltarget_returntype_getdefault_name.data(), signature,
+      signatureLength, &callTargetReturnTypeGetDefault);
+  if (FAILED(hr)) {
+    Warn("Wrapper callTargetReturnTypeGetDefault could not be defined.");
+    return mdMemberRefNil;
+  }
+
+  return callTargetReturnTypeGetDefault;
+}
+
+mdMethodSpec CallTargetTokens::GetCallTargetDefaultValueMethodSpec(
+    FunctionMethodArgument* methodArgument) {
+  auto hr = EnsureBaseCalltargetTokens();
+  if (FAILED(hr)) {
+    return mdMethodSpecNil;
+  }
+
+  mdMethodSpec getDefaultMethodSpec = mdMethodSpecNil;
+  ModuleMetadata* module_metadata = GetMetadata();
+
+  // *** Ensure we have the CallTargetInvoker.GetDefault<> memberRef
```

Comment:
```suggestion
  // *** Ensure we have the CallTargetInvoker.GetDefaultValue<> memberRef
```

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/1007/commits/9e7bcb3955022de47db4e80e29c33db9f64b69c2